src/zenml/config/global_config.py

```diff
@@ -566,6 +566,8 @@ def get_default_store(self) -> StoreConfiguration:
             config = StoreConfiguration(
                 **env_store_config,
             )
+        elif self.store:
+            config = self.store
```

Comment:
To add to Michael's comment: it seems to me that in this case either ENV go or File, but not a merger of them, isn't it?

Reply:
That's a good point! maybe I understood the ticket differently. In that case, I imagine we can do something like the following, where we see if self.config is set and then update config based on existing env variables. @schustmi @avishniakov 

```
# Update self.config based on environment variables
if self.config:
    # Update the store configuration from environment variables
    if env_store_config:
        if "type" not in env_store_config and "url" in env_store_config:
            env_store_config["type"] = BaseZenStore.get_store_type(env_store_config["url"])

        # Merge environment variables into self.config for store
        for key, value in env_store_config.items():
            self.config[key] = value

    # Update the secrets store configuration
    for key, value in env_secrets_store_config.items():
        self.config.secrets_store[key] = value

    # Update the backup secrets store configuration
    for key, value in env_backup_secrets_store_config.items():
        self.config.backup_secrets_store[key] = value

    logger.debug("Updated the configuration with environment variables.")
else:
    if len(env_store_config):
        if "type" not in env_store_config and "url" in env_store_config:
            env_store_config["type"] = BaseZenStore.get_store_type(env_store_config["url"])

        logger.debug("Using environment variables to configure the default store")

        self.config = StoreConfiguration(**env_store_config)
    else:
        self.config = BaseZenStore.get_default_store_config(
            path=os.path.join(self.local_stores_path, DEFAULT_STORE_DIRECTORY_NAME)
        )
```

Reply:
Yeah, something like this might do. Can you write some tests to cover this use-case?

Reply:
@avishniakov Looking back on this, we can't merge. With the env variables, we need to initialize a new zen store which in this case will be the `RestZenStore` if we use the already existing one which is SQLZenStore config and we just updated (merge) to initialize the rest one it will fail because [rest zen store](https://github.com/zenml-io/zenml/blob/develop/src/zenml/zen_stores/rest_zen_store.py#L255) doesn't the same attributes as [sql one](https://github.com/zenml-io/zenml/blob/develop/src/zenml/zen_stores/sql_zen_store.py#L352) and each of these stores has different attributes besides type, URL, and SSL.

Reply:
I implemented a custom merge logic that in some cases replaces the config file store completely, in other cases it only overrides some of its attributes.